### PR TITLE
HIVE-28282: Addendum: Merge-Insert with reserved keywords as column names

### PR DESCRIPTION
**Pull request by deniskuzZ**
### What changes were proposed in this pull request?

Merge-Insert with reserved keywords as column names

### Why are the changes needed?

Fix

### Does this PR introduce _any_ user-facing change?

No
### Is the change a dependency upgrade?

No

### How was this patch tested?

Qtest

### CODE REVIEW

File: ql/src/java/org/apache/hadoop/hive/ql/parse/rewrite/sql/MultiInsertSqlGenerator.java [hunk 1]
```diff
@@ -155,22 +156,22 @@ public void appendPartColsOfTargetTableWithComma(String alias) {
...
-
-  public void appendCols(List<FieldSchema> columns) {
-    appendCols(columns, null, null);
+  
+  public <T> void appendCols(List<T> columns, Function<T, String> func) {
+    appendCols(columns, null, null, func);
   }
 
-  public void appendCols(List<FieldSchema> columns, String alias, String prefix) {
+  public <T> void appendCols(List<T> columns, String alias, String prefix, Function<T, String> func) {
```
Comment (kasakrisz) [inline ql/src/java/org/apache/hadoop/hive/ql/parse/rewrite/sql/MultiInsertSqlGenerator.java:174]: nit. rename `func` to `stringConverter`

**Review by kasakrisz — APPROVED** [review]
+1

**Comment by kasakrisz** [comment]
Please add the original commit message of [HIVE-28282](https://issues.apache.org/jira/browse/HIVE-28282) too.

File: ql/src/java/org/apache/hadoop/hive/ql/parse/rewrite/sql/MultiInsertSqlGenerator.java (hunk 1 above)
Comment (deniskuzZ) [reply ql/src/java/org/apache/hadoop/hive/ql/parse/rewrite/sql/MultiInsertSqlGenerator.java]: renamed

**Comment by deniskuzZ** [comment]
> Please add the original commit message of [HIVE-28282](https://issues.apache.org/jira/browse/HIVE-28282) too.

I am not sure where I should add it?

**Pull request merged**